profiles/common/modules/features/nexteuropa_dgt_connector/nexteuropa_dgt_connector.module

```diff
@@ -3,3 +3,59 @@
  * @file
  * Code for the DGT connector feature.
  */
+
+define('NEXTEUROPA_DGT_CONNECTOR_NEXTEUROPA_MAX_SMALLJOB_LENGTH', 300);
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * Change the form on the 'Translate' tab to provide the tmgmt_poetry module
+ * functionalities.
+ */
+function nexteuropa_dgt_connector_form_tmgmt_entity_ui_translate_form_alter(&$form, &$form_state) {
+
+  $entity = $form_state['entity'];
+  $length = _nexteuropa_dgt_connector_get_content_length($entity);
+
+  if ($length > NEXTEUROPA_DGT_CONNECTOR_NEXTEUROPA_MAX_SMALLJOB_LENGTH) {
+    // Remove add to cart button.
+    unset($form['top_actions']);
+  }
+
+  // Set plugin for #cart .
+  if (isset($form_state['tmgmt_cart'])) {
+    $form_state['tmgmt_cart']['plugin'] = 'workbench_moderation';
+  }
+}
+
+/**
+ * Calculate the total length of the translatable strings of a content.
+ *
+ * @param object $entity
+ *    The entity to translate.
+ *
+ * @return int
+ *    Length of all translatable content.
+ */
+function _nexteuropa_dgt_connector_get_content_length($entity) {
+
+  $length = 0;
+  $instances = field_info_instances($entity->entity_type, $entity->type);
```

Comment:
Here I would suggest using not very well documented but useful function.
```
$fields = field_read_fields(array(
  'entity_type' => 'node',
  'bundle' => 'page',
  'module' => 'text',
  'translatable' => TRUE,
));
```
In that way you will have all of the fields in one shot.